biorxiv/corpora_comparison/biorxiv_pmc_corpora_comparison.py

```diff
@@ -0,0 +1,445 @@
+            del corpus_one[stopword]
+
+        if stopword in corpus_two:
+            del corpus_two[stopword]
+            
+    return corpus_one, corpus_two
+
+
+# # Full Text Comparison (Global)
+
+# ## Gather Word Frequencies
+
+# In[4]:
+
+
+biorxiv_corpus_count = (
+    aggregate_word_counts(
+        list(Path("output/biorxiv_word_counts").rglob("*tsv"))
+    )
+)
+
+
+# In[5]:
+
+
+pmc_corpus_count = (
+    aggregate_word_counts(
+        list(Path("output/pmc_word_counts").rglob("*tsv"))
+    )
+)
+
+
+# In[8]:
+
+
+biorxiv_corpus_count.most_common(10)
+
+
+# In[9]:
+
+
+pmc_corpus_count.most_common(10)
+
+
+# In[6]:
+
+
+pickle.dump(biorxiv_corpus_count, open("output/biorxiv_total_count.pkl", "wb"))
+pickle.dump(pmc_corpus_count, open("output/pmc_total_count.pkl", "wb"))
+
+
+# ## Analysis without Stop Words
+
+# In[5]:
+
+
+biorxiv_corpus_count = pickle.load(open("output/biorxiv_total_count.pkl", "rb"))
+pmc_corpus_count = pickle.load(open("output/pmc_total_count.pkl", "rb"))
+
+
+# In[6]:
+
+
+biorxiv_corpus_count, pmc_corpus_count = remove_stop_words(
+    biorxiv_corpus_count,
+    pmc_corpus_count
+)
+
+
+# In[7]:
+
+
+top_ten_biorxiv = biorxiv_corpus_count.most_common(100)
+top_ten_biorxiv[0:10]
+
+
+# In[8]:
+
+
+top_ten_pmc = pmc_corpus_count.most_common(100)
+top_ten_pmc[0:10]
+
+
+# In[9]:
+
+
+print("Number of words in biorxiv but not in Pubmed Central:")
+biorxiv_difference = set(list(biorxiv_corpus_count.keys())) - set(list(pmc_corpus_count.keys()))
+print(len(biorxiv_difference))
+
+
+# In[10]:
+
+
+[
+    biorxiv_difference.pop()
+    for i in range(10)
+]
+
+
+# In[11]:
+
+
+print("Number of words in Pubmed Central but not in biorxiv:")
+pmc_difference = set(list(pmc_corpus_count.keys())) - set(list(biorxiv_corpus_count.keys()))
+print(len(pmc_difference))
+
+
+# In[12]:
+
+
+[
+    pmc_difference.pop()
+    for i in range(10)
+]
+
+
+# In[13]:
+
+
+total_words = set(list(dict(top_ten_biorxiv).keys()) + list(dict(top_ten_pmc).keys()))
+data = []
+for word in tqdm_notebook(total_words):
+    
+    word_stat = get_term_statistics(
+        biorxiv_corpus_count,
+        pmc_corpus_count,
+        word
+    )
+    
+    data.append({
+        "lemma": word,
+        "biorxiv_count":biorxiv_corpus_count[word] if word in biorxiv_corpus_count else 0,
+        "pmc_count":pmc_corpus_count[word] if word in pmc_corpus_count else 0,
+        "biorxiv_total":word_stat['chi_sq'][3][1,0],
+        "pmc_total":word_stat['chi_sq'][3][1,1],
+        "log_p": word_stat['chi_sq'][1],
+        "log_likelihood": word_stat['log_likelihood'],
+        "odds_ratio": word_stat['odds_ratio']
+    })
+
+
+# In[14]:
+
+
+total_word_stats_df = pd.DataFrame.from_records(data)
+total_word_stats_df.to_csv(
+    "output/full_corpus_comparison_stats.tsv", 
+    sep="\t", index=False
+)
+total_word_stats_df.head()
+
+
+# In[15]:
+
+
+(
+    total_word_stats_df
+    .sort_values("log_likelihood", ascending=False)
+    .head(20)
+)
```

Comment:
All the p_values in this cell and the previous cell are the same. I assume that they're just the log of your epsilon value, but it may be worth checking.

Reply:
Yeah. The chi-squared test breaks when numbers reach epic proportions. As a consequence everything becomes "significant", which makes hypothesis testing pretty difficult. This is also why I added log-likelihood and odds ratio calculations.